juspay/hyperswitch
# refactor: create separate struct for surcharge details response

## Type of Change



## Description

Changes in this PR:
1. Renamed and restructured a few existing enums and structs
2. Introduced new `SurchargeDetails` struct for use in core and `SurchargeDetailsResponse` struct for api response.
3. Surcharge details validation during payment_confirm shall only be made if surcharge_details is sent during confirm call. Not otherwise.
4. Added amount validation is payment create and payment update flow, to cover surcharge.


### Additional Changes





## Motivation and Context



## How did you test it?

Steps to test:
1. Create a merchant with paypal connector
2. Configure some surcharge rules for the merchant
3. Create a payment
4. Do merchant payment_method list. (should get below body in `surcharge_details` field)
```
"surcharge_details": {
    "surcharge": {
        "type": "fixed",
        "value": 123
    },
    "tax_on_surcharge": {
        "percentage": 5.09
    },
    "display_surcharge_amount": 1.23,
    "display_tax_on_surcharge_amount": 0.07,
    "display_final_amount": 66.7
},
```


## Checklist
